Trace 0:
  Transitions:
    'when PyObject_GetItem() succeeds'
    'returning'
  Return value:
    repr(): PointerToRegion(gcctype='struct PyObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=29), region=RegionOnHeap('new ref from call to PyObject_GetItem', gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=29)))
    str(): (struct PyObject *)&RegionOnHeap('new ref from call to PyObject_GetItem', gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=29)) from tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c:29
    r->ob_refcnt: refs: 1 owned
    r->ob_type: PointerToRegion(gcctype='struct PyTypeObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=29), region=Region('PyTypeObject for new ref from call to PyObject_GetItem'))
  region for gcc.ParmDecl('self') on stack:
    repr(): RegionOnStack("region for gcc.ParmDecl('self')")
    str(): region for gcc.ParmDecl('self') on stack
    r->ob_refcnt: refs: 0 owned, 1 borrowed
    r->ob_type: PointerToRegion(gcctype='struct PyTypeObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=27), region=Region("region-for-type-of-arg-gcc.ParmDecl('self')"))
  region for gcc.ParmDecl('key') on stack:
    repr(): RegionOnStack("region for gcc.ParmDecl('key')")
    str(): region for gcc.ParmDecl('key') on stack
    r->ob_refcnt: refs: 0 owned, 1 borrowed
    r->ob_type: PointerToRegion(gcctype='struct PyTypeObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=27), region=Region("region-for-type-of-arg-gcc.ParmDecl('key')"))
  Exception:
    (struct PyObject *)0 from tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c:28

Trace 1:
  Transitions:
    'when PyObject_GetItem() fails'
    'returning'
  Return value:
    repr(): ConcreteValue(gcctype='struct PyObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=29), value=0)
    str(): (struct PyObject *)0 from tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c:29
  region for gcc.ParmDecl('self') on stack:
    repr(): RegionOnStack("region for gcc.ParmDecl('self')")
    str(): region for gcc.ParmDecl('self') on stack
    r->ob_refcnt: refs: 0 owned, 1 borrowed
    r->ob_type: PointerToRegion(gcctype='struct PyTypeObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=27), region=Region("region-for-type-of-arg-gcc.ParmDecl('self')"))
  region for gcc.ParmDecl('key') on stack:
    repr(): RegionOnStack("region for gcc.ParmDecl('key')")
    str(): region for gcc.ParmDecl('key') on stack
    r->ob_refcnt: refs: 0 owned, 1 borrowed
    r->ob_type: PointerToRegion(gcctype='struct PyTypeObject *', loc=gcc.Location(file='tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c', line=27), region=Region("region-for-type-of-arg-gcc.ParmDecl('key')"))
  Exception:
    (struct PyObject *)&RegionForGlobal(gcc.VarDecl('PyExc_MemoryError')) from tests/cpychecker/refcounts/PyObject_GetItem/correct/input.c:29
